src/SimpleWMSBody.cpp

```diff
@@ -0,0 +1,552 @@
+////////////////////////////////////////////////////////////////////////////////////////////////////
+//                               This file is part of CosmoScout VR                               //
+//      and may be used under the terms of the MIT license. See the LICENSE file for details.     //
+//                        Copyright: (c) 2019 German Aerospace Center (DLR)                       //
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#include "SimpleWMSBody.hpp"
+
+#include "../../../src/cs-core/GraphicsEngine.hpp"
+#include "../../../src/cs-graphics/TextureLoader.hpp"
+#include "../../../src/cs-utils/FrameTimings.hpp"
+#include "../../../src/cs-utils/filesystem.hpp"
+#include "../../../src/cs-utils/utils.hpp"
+
+#include <curlpp/Infos.hpp>
+#include <curlpp/Options.hpp>
+
+namespace csp::simplewmsbodies {
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+const uint32_t GRID_RESOLUTION_X = 200;
+const uint32_t GRID_RESOLUTION_Y = 100;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+const std::string SimpleWMSBody::SPHERE_VERT = R"(
+uniform vec3 uSunDirection;
+uniform vec3 uRadii;
+uniform mat4 uMatModelView;
+uniform mat4 uMatProjection;
+
+// inputs
+layout(location = 0) in vec2 iGridPos;
+
+// outputs
+out vec2 vTexCoords;
+out vec3 vPosition;
+out vec3 vCenter;
+out vec2 vLonLat;
+
+const float PI = 3.141592654;
+
+void main()
+{
+    vTexCoords = vec2(iGridPos.x, 1-iGridPos.y);
```

Comment:
check white spaces according to our clang-format

vec2(iGridPos.x, 1 - iGridPos.y);